### PR TITLE
fix: only show FileOp autoType input for csv/tsv formats

### PR DESCRIPTION
**Pull request by akre54**
## Summary
- `autoType` now only appears in the FileOp UI when `format` is `csv` or `tsv`, hidden otherwise.
- Fixes a bug in the original approach: routing the toggle through `addCustomInput`/`removeCustomInput` (with the subscription set up inside `createInputs()`) caused an infinite `rebuildInputs()` recursion — every rebuild recreated the `format` field, whose fresh subscription fired immediately and re-triggered the rebuild. This crashed the browser tab as soon as `format` was set to `csv`.
- Fix: keep `autoType` as a normal, always-present input (as before) and toggle its visibility once via the existing `showField`/`hideField`/`isFieldVisible` mechanism (same one used for hiding `pulse` by default), subscribing in the constructor rather than inside `createInputs()`.
- Also fixes a non-existent `toHaveKey` matcher in the new test (real one is `toHaveProperty`), and de-duplicates a reused test id.

## Test plan
- [x] `npm test src/noodles/operators.test.ts` — 293/293 passing
- [x] Manually reproduced the crash on the prior version by calling `operator.inputs.format.setValue('csv')` in a scratch test; confirmed fixed on this branch
- [ ] Load a project with a FileOp node, switch format between json/csv/tsv/text/binary in the browser and confirm `autoType` shows only for csv/tsv

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: noodles-editor/src/noodles/operators.ts [hunk 1]
```diff
@@ -1903,13 +1903,22 @@ export class FileOp extends Operator<FileOp> {
     'Fetch a file from a URL or text. Supports csv, tsv, json, text, and binary formats'
   asDownload = () => this.outputData
 
+  constructor(id: OpId, inputs?: unknown, locked?: boolean) {
+    super(id, inputs, locked)
+
+    const sub = this.inputs.format.subscribe(format => {
+      this.setFieldVisibility('autoType', format === 'csv' || format === 'tsv')
+    })
```
[the remark this answers is not in the document]
Comment (chrisgervang) [reply noodles-editor/src/noodles/operators.ts:1911]: Seems worth looking into

**Review by chrisgervang — APPROVED** [review]
